lib/puppet/provider/service/service.rb

```diff
@@ -36,6 +36,9 @@ def ucommand(type, fof = true)
     else
       cmd = [send("#{type}cmd")].flatten
     end
+    if (( type =~ /^start/ ) && ( Facter.value(:osfamily) == "Solaris" ))
+      cmd = ["/usr/bin/ctrun -l none", cmd].flatten.join(" ")
+    end
```

Comment:
Updated PR with the ```texecute``` method override in ```init```. (Since management of Solaris init.d services on apparently requires explicitly declaring "provider=init", it looks like this needed to be done in ```init``` instead of ```smf```.)